### PR TITLE
feat(newsletter): added in-memory driver for newsletter lib

### PR DESCRIPTION
**Pull request by Nolan-Arnold**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/newsletter/testing/src/drivers/in-memory/newsletter-driver.module.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { CommonModule } from "@angular/common";
+import { DaffNewsletterDriver } from "libs/newsletter/src/driver/injection-tokens/newsletter-driver.token";
```
Comment (damienwebdev) [inline libs/newsletter/testing/src/drivers/in-memory/newsletter-driver.module.ts]: Path

File: libs/newsletter/testing/src/drivers/in-memory/newsletter.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpTestingController, HttpClientTestingModule } from '@angular/common/http/testing';
+
+import { DaffInMemoryNewsletterService } from './newsletter.service';
+import { DaffNewsletterSubmission } from '@daffodil/newsletter';
+import { DaffNewsletterUnion } from 'libs/newsletter/src/models/newsletter-union';
```
Comment (damienwebdev) [inline libs/newsletter/testing/src/drivers/in-memory/newsletter.service.spec.ts]: Path

File: libs/newsletter/testing/src/drivers/in-memory/newsletter.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * The newsletter inmemory driver to mock the newsletter backend service.
+ * 
+ * @Param HttpClient
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffInMemoryNewsletterService implements DaffNewsletterServiceInterface<DaffNewsletterUnion, DaffNewsletterUnion>{
```
Comment (damienwebdev) [inline libs/newsletter/testing/src/drivers/in-memory/newsletter.service.ts]: This generic usage is pretty neat.

File: libs/newsletter/testing/src/drivers/in-memory/newsletter.service.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * @Param HttpClient
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffInMemoryNewsletterService implements DaffNewsletterServiceInterface<DaffNewsletterUnion, DaffNewsletterUnion>{ 
+  constructor() {}
+
+  send(email: DaffNewsletterUnion): Observable<DaffNewsletterUnion> {
+    return of(email);
```
Comment (damienwebdev) [inline libs/newsletter/testing/src/drivers/in-memory/newsletter.service.ts]: Can we have a way that we can specifically cause it to fail for testing purposes?

File: libs/newsletter/testing/src/drivers/in-memory/newsletter.service.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+
+import { DaffNewsletterServiceInterface, DaffNewsletterSubmission } from '@daffodil/newsletter';
+import { DaffNewsletterUnion } from 'libs/newsletter/src/models/newsletter-union';
```
Comment (damienwebdev) [inline libs/newsletter/testing/src/drivers/in-memory/newsletter.service.ts]: Path

File: libs/newsletter/testing/src/drivers/in-memory/newsletter.service.ts (hunk 4 above)
Comment (Nolan-Arnold) [reply libs/newsletter/testing/src/drivers/in-memory/newsletter.service.ts]: Okay, I tried to address this by having it throw an error with an undefined payload.

Comment (damienwebdev) [reply libs/newsletter/testing/src/drivers/in-memory/newsletter.service.ts]: That'll work.

**Comment by damienwebdev** [comment]
@Nolan-Arnold there's a minor merge conflict, otherwise this PR lgtm.

File: libs/newsletter/src/effects/newsletter.effects.spec.ts [hunk 6]
```diff
@@ -11,9 +11,9 @@ import { DaffTestingNewsletterService } from '@daffodil/newsletter/testing';
 
 describe('NewsletterEffects', () => {
   let actions$: Observable<any>;
-  let effects: DaffNewsletterEffects<DaffNewsletterSubmission, any>;//come back to this
-  const mockNewsletter = {email: 'test@test.com'};
-  let daffNewsletterDriver: DaffNewsletterServiceInterface<DaffNewsletterSubmission, any>;//and this
+  let effects: DaffNewsletterEffects<DaffNewsletterSubmission, any>;
+  const mockNewsletter = {email: "test@test.com"};
```
Comment (damienwebdev) [inline libs/newsletter/src/effects/newsletter.effects.spec.ts]: Should be single quotes.

**Pull request merged**